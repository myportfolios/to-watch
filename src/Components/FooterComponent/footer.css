.footer {
  padding: 0 3rem;
  background-color: #ff6347;
  display: flex;
  flex-direction: column;
  width: 100vw;
  align-items: center;
  bottom: 0;
  height: 460px;
}
.footer-popup {
  width: 700px;
  height: 500px;
  position: absolute;
  transform: translate(-50%, -50%);
  left: 50%;
  top: 50%;
  box-shadow: 0 5px 10px #eee;
}

.footer-header-box {
  align-items: center;
}
.footer-header {
  padding: 3rem 0 1.5rem 0;
}
.footer-text {
  text-align: center;
}
#footer-newsletter {
  text-align: center;
  border: 1px solid;
  width: 30%;
  margin: 0 auto;
  padding: 0.5rem 0;
  color: #ff6347;
  background-color: white;
  border-radius: 90px;
  text-transform: uppercase;
  font-size: larger;
  letter-spacing: 0.2rem;
  margin-bottom: 1.5rem;
  margin-top: 1.5rem;
}
#footer-newsletter:hover {
  background-color: #0000ff;
  /* color: white; */
  font-size: x-large;
  padding: 0.5rem 0;
}
.links {
  padding: 0 2rem;
}
.footer-subscribe-box {
  border: 1px solid white;
  padding: 1rem 0;
  margin: 1rem 0;
  width: inherit;
}
#email-subscribe-field {
  width: 260px;
  position: relative;
  transform: translate(-50%, -50%);
  left: 50%;
  padding: 10px 0;
  top: 35px;
  border-radius: 9px;
  -webkit-transition: width 0.4s ease-in-out;
  transition: width 0.4s ease-in-out;
}
#email-subscribe-field:focus {
  background-color: #eee;
  width: 70%;
}
