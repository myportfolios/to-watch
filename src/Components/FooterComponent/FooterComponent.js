import React from "react";
import { Link, withRouter } from "react-router-dom";
import "./footer.css";

function FooterComponent(props) {
  const { location } = props;

  return (
    <>
      {location.pathname !== "/contact" ? (
        <div className="footer">
          <div className="footer-header-box">
            <h3 className="footer-header">
              Would you like to get the latest movies in your inbox?
            </h3>
            <h5 className="footer-text">
              Subscribe for our weekly newsletter below
            </h5>
            <div className="footer-subscribe-box">
              <input placeholder="Email" id="email-subscribe-field" />
              <p id="footer-newsletter">Subscribe</p>
            </div>
          </div>
          <div className="footer-links">
            <Link to="./home" className="links">
              Home
            </Link>
            <Link to="./about" className="links">
              About
            </Link>
            <Link to="./trendy-movies" className="links">
              Trendy Movies
            </Link>
            <Link to="./search-movie" className="links">
              Search Movie
            </Link>
            <Link to="./contact" className="links">
              Contact
            </Link>
          </div>
        </div>
      ) : (
        <FooterContactBox />
      )}
    </>
  );
}
export default withRouter(FooterComponent);

const FooterContactBox = () => {
  return (
    <div className="footer-popup">
      <div className="footer-header-box">
        <h3 className="footer-header">
          Would you like to get the latest movies in your inbox?
        </h3>
        <h5 className="footer-text">
          Subscribe for our weekly newsletter below
        </h5>
        <div className="footer-subscribe-box">
          <input placeholder="Email" id="email-subscribe-field" />
          <p id="footer-newsletter">Subscribe</p>
        </div>
      </div>
    </div>
  );
};
