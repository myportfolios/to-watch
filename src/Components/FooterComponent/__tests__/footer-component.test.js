import React from "react";
import { shallow } from "enzyme";
import { configure } from "enzyme";

import Adapter from "enzyme-adapter-react-16";
import FooterComponent from "../FooterComponent";

configure({ adapter: new Adapter() });

describe("FooterComponent", () => {
  let props;
  const wrapper = shallow(<FooterComponent {...props} />);
  it("renders correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
