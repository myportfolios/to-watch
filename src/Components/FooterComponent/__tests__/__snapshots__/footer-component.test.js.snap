// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FooterComponent renders correctly 1`] = `ShallowWrapper {}`;
